sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -1130,78 +1039,91 @@ class Std {
     builtinWithDefaults("setUnion", "a" -> null, "b" -> null, "keyF" -> Val.False(dummyPos)) { (args, pos, ev) =>
       val a = args(0) match {
         case arr: Val.Arr => arr.asLazyArray
-        case str: Val.Str => stringChars(pos, str.value).asLazyArray
+        case str: Val.Str if !ev.settings.strictSetOperations => stringChars(pos, str.value).asLazyArray
         case _ => Error.fail("Arguments must be either arrays or strings")
```

Comment:
I think this error message, as written, is only valid for the old behavior. With strict set operations enabled, I think it should read `"Arguments must be arrays"`.

I see that this

```scala
match {
        case arr: Val.Arr => arr.asLazyArray
        case str: Val.Str if !ev.settings.strictSetOperations => stringChars(pos, str.value).asLazyArray
        case _ => Error.fail("Arguments must be either arrays or strings")
      }
```

style of block is pre-existingly copy-paste duplicated across all of the set operation implementations.

WDYT about factoring it out into a helper function and then cleaning up the error message to respect the new flag?